src/component.tsx

```diff
@@ -0,0 +1,479 @@
+	]
+
+	static defaultProps = {
+		theme: 'snow',
+		modules: {},
+		readOnly: false,
+	}
+
+	state: ReactQuillState = {
+		generation: 0,
+		selection: null,
+		value: '',
+	}
+
+	/*
+	The Quill Editor instance.
+	*/
+	editor?: Quill
+
+	/*
+	Reference to the element holding the Quill editing area.
+	*/
+	editingArea?: React.ReactInstance | null
+
+	/*
+	Used to compare whether deltas from `onChange` are being used as `value`.
+	*/
+	lastDeltaChangeSet?: DeltaStatic
+
+	/*
+	Stores the contents of the editor to be restored after regeneration.
+	*/
+	regenerationSnapshot?: {
+		delta: DeltaStatic,
+		selection: Range,
+	}
+
+	constructor(props: ReactQuillProps) {
+		super(props);
+		const value = this.isControlled()? props.value : props.defaultValue;
+		this.state.value = value ?? '';
+	}
+
+	validateProps(props: ReactQuillProps): void {
+		if ('toolbar' in props) throw new Error(
+			'The `toolbar` prop has been deprecated. Use `modules.toolbar` instead. ' +
+			'See: https://github.com/zenoamaro/react-quill#upgrading-to-react-quill-v100'
+		);
+
+		if (props.modules?.toolbar?.[0]?.type) throw new Error(
+			'Since v1.0.0, React Quill will not create a custom toolbar for you ' +
+			'anymore. Create a toolbar explictly, or let Quill create one. ' +
+			'See: https://github.com/zenoamaro/react-quill#upgrading-to-react-quill-v100'
+		);
+
+		if (props.formats && (
+			!(props.formats instanceof Array) ||
+			some(props.formats, x => typeof x !== 'string')
+		)) throw new Error(
+			'You cannot specify custom `formats` anymore. Use Parchment instead.  ' +
+			'See: https://github.com/zenoamaro/react-quill#upgrading-to-react-quill-v100.'
+		);
+
+		if ('styles' in props) throw new Error(
+			'The `styles` prop has been deprecated. Use custom stylesheets instead. ' +
+			'See: https://github.com/zenoamaro/react-quill#upgrading-to-react-quill-v100.'
+		);
+
+		if ('pollInterval' in props) throw new Error(
+			'The `pollInterval` property does not have any effect anymore. ' +
+			'Remove it from your props. ' +
+			'See: https://github.com/zenoamaro/react-quill#upgrading-to-react-quill-v100.'
+		);
+
+		if (React.Children.count(props.children) > 1) throw new Error(
+			'The Quill editing area can only be composed of a single React element.'
+		);
+
+		if (React.Children.count(props.children)) {
+			const child = React.Children.only(props.children);
+			if (child?.type === 'textarea') throw new Error(
+				'Quill does not support editing on a <textarea>. Use a <div> instead.'
+			);
+		}
+
+		if (
+			this.lastDeltaChangeSet &&
+			props.value === this.lastDeltaChangeSet
+		) throw new Error(
+			'You are passing the `delta` object from the `onChange` event back ' +
+			'as `value`. You most probably want `editor.getContents()` instead. ' +
+			'See: https://github.com/zenoamaro/react-quill#using-deltas'
+		);
+	}
+
+	shouldComponentUpdate(nextProps: ReactQuillProps, nextState: ReactQuillState) {
+		// TODO: Is there a better place to validate props?
```

Comment:
I'll try this out.

Reply:
Just realized that `componentDidUpdate` does not always get called, as we prevent many updates. Render is the other choice, but it would have the same effect. So `shouldComponentUpdate` seems the only good place.

However, I also realized that I have broken the test for this feature 😬.

Reply:
Test is fixed.